### PR TITLE
Implement easing 

### PR DESCRIPTION
**Pull request by foldfelis**

### CODE REVIEW

**Comment by foldfelis** [comment]
@chengchingwen please check correctness of the implementation

**Comment by foldfelis** [comment]
This PR will remains a draft before the test is added.

**Comment by foldfelis** [comment]
[Easing example](https://github.com/raysan5/raylib/blob/master/examples/others/easings_testbed.c)

File: src/easing.jl [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+    return -c/2 * ((t - 1) * (t - 3) - 1) + b
+end
+
+# Exponential Easing functions
+function EaseExpoIn(t::Real, b::Real, c::Real, d::Real)
+    return iszero(t) ? b : c * 2^(10(t/d - 1)) + b
+end
+function EaseExpoOut(t::Real, b::Real, c::Real, d::Real)
+    return iszero(t-d) ? b+c : c * (-(2.0^(-10 * t/d)) + 1) + b
```
Comment (chengchingwen) [inline src/easing.jl:87]:
```suggestion
    return t ≈ d ? b+c : c * (-(2.0^(-10 * t/d)) + 1) + b
```

Generally this kind of check should use `isapprox` due to floating point precision.

File: src/easing.jl [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+# Linear Easing functions
+EaseLinearNone(t::Real, b::Real, c::Real, d::Real) = c * t/d + b
+EaseLinearIn(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+EaseLinearOut(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+EaseLinearInOut(t::Real, b::Real, c::Real, d::Real) = EaseLinearNone(t, b, c, d)
+
+# Sine Easing functions
+EaseSineIn(t::Real, b::Real, c::Real, d::Real) = -c * cos(t/d * (π/2)) + c + b
+EaseSineOut(t::Real, b::Real, c::Real, d::Real) = c * sinf(t/d * (π/2)) + b
```
Comment (chengchingwen) [inline src/easing.jl:18]:
```suggestion
EaseSineOut(t::Real, b::Real, c::Real, d::Real) = c * sin(t/d * (π/2)) + b
```

File: src/easing.jl [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    return c/2 * (sqrt(1 - t^2) + 1) + b
+end
+
+# Cubic Easing functions
+function EaseCubicIn(t::Real, b::Real, c::Real, d::Real)
+    t /= d
+
+    return c * t^3 + b
+end
+function EaseCubicOut(t::Real, b::Real, c::Real, d::Real)
```
Comment (chengchingwen) [inline src/easing.jl:48]: need newline between functions

File: src/easing.jl [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        return EaseBounceIn(2t, 0, c, d)/2 + b
+    end
+
+    return EaseBounceOut(2t-d, 0, c, d)/2 + c/2 + b
+end
+
+# Elastic Easing functions
+function EaseElasticIn(t::Real, b::Real, c::Real, d::Real)
+    iszero(t) && (return b)
+    isone(t/=d) && (return b + c)
```
Comment (chengchingwen) [inline src/easing.jl:157]: also this

**Review by chengchingwen — CHANGES_REQUESTED** [review]

**Comment by chengchingwen** [comment]
This didn't seem to be using any raymath apis, should be separate to different PR. And under this situation we can actually merge the easing part into master branch since there are no dependencies at all.

File: src/easing.jl (hunk 2 above)
Comment (chengchingwen) [reply src/easing.jl:18]: We would need a more thorough test to avoid this kind of error

**Comment by foldfelis** [comment]
move to #23

**Pull request closed without merging**